### ISSUE TITLE
Girls need some improvements of KVIrc ! Please help us to stay on chats by integrating at least a button to block private messages ! Sexual harassment is a reality and we need more ergonomic ban functions ! Here is also a kvirc tutorial in french.

### ISSUE THREAD

**Issue by Pandore-II** [kvirc/KVIrc#2653]
Thanks a lot for Kvirc. Some months ago, I discovered the IRC chat.
I would like to share my experience and the problems I have noticed.
I started with chaat.fr and europnet.org

I often wanted to leave the chat. Fortunately I met some chatfriends who understood and helped me. Thanks to Pyro, pontdesarts, Dart, Metro, _A_vos_pied and so much other. There are a few good men in the chat but an overwhelming majority of frustrated men !

On these websites, we can chat but women get so much private messages. The harassment, and especially the sexual harassment is very hard for the spirit.
Month after month, I tried to find solutions to block bad guys. On websites chaat.fr and europnet.org , every day, the list of ignored users is reseted. So, every day, the hell is reopened.

So, I tried hexchat... then Quassel... When we get a lot of private messages, we need history because we can not remember all chats.
hexchat is not ergonomic. For instance, we can not reload the history of private messages after the user has left.

Quassel crashed after 1000 blocked mens. I know, this is a lot of people... too much for Quassel.

Finally, I discovered KVirc and Andaris. This software is very powerful. Thanks a lot for you work.
However, there is a lake of ergonomics because it has not been thought for mass blocking... so, it is not adapted for women !

That is the reason why I came here to highlight some improvements. I hope that my ideas will be received positively, even if women are not the main users of the IRC chat.

First of all, there are not tutorial in french (in kvric and in Andaris websites). Please, find my tutorial in attachement. In the absence of another, I suggest you publish it here : https://www.kvirc.net/?id=tutorials&lang=en It can sound a little feminist. I am sorry for that.

I noticed several problems in Kvirc :

1) There is not button to block a private message. I do not want to block all private messages, but just selectively... and quickly !
Furthermore, the "ignore" command will block users in channel... They can defame me or insult me on channel because mens are hurt in their ego of having be blocked in private.
So, I must be able to see that in channel and to defend me.
In my tutorial, I explained how to create a block button in the toolbars. We need this in all Kvirc, by default. We also need another button for the "ignore" command.
And, another button to complain to the chat moderator. As you can see, my button will report the harasser on the #metoo channel.
I would like to copy-past his message but I did not find how to do.

Here is the program of my blocking button :

>    say Allez, va faire un peu de Kung Fu https://www.youtube.com/watch?v=g2c_-9kIjyU , ça calme l'égo des mâles en rut !
>    say $target vient d'être signalé par $me sur #metoo pour message privé non sollicité.
>    invite $target #metoo
>    PRIVMSG #metoo :$target ($realname($target)) vient d'être bloqué par $me pour message privé non sollicité : $context.lastMessageTime
>    ignore $target
>    reguser.setIgnoreFlags -q -p -n -t -i -d $target
>    window.close

   
   Attached pictures show the button in the toolsbar and the report of harassment on the channel #metoo.

![Btn](https://github.com/kvirc/KVIrc/assets/172998160/85798c80-3cbe-4843-9969-80c388cc0d7d)

2) When we right click a user on the list of private message, we can only close the message... the block button should also appear.
In the list of users, on a channel, there is an "ignore" button but not for private message.
So, when I get a private message, I have to search the channel who the user is, to search the user in the users list of the channel to click "ignore" on him.
So much time !!

   The attached picture shows and exemple of sexual harassment : A guy asked to see my tits !  There is no button to block him in the right click menu !

![seins](https://github.com/kvirc/KVIrc/assets/172998160/a797a367-6bc7-4fd6-992a-575fc22215db)

Several reports on #metoo :

![Report](https://github.com/kvirc/KVIrc/assets/172998160/1e64d8d2-5fe6-45f4-9cf2-87613a0ef6aa)

3) Blocked users are added in the list of "registered user".
Sorry to say it but this list is abominable.

Please, think how to unblock a user if you have 500 users in this list ?

![PasDeRecherche](https://github.com/kvirc/KVIrc/assets/172998160/ebb05e94-99a5-44d7-b7e3-56b1d2fed92b)

There is not search function... The list is not alphabetically sorted.
Oh my god ! I am affraid to have to unblock a user.
4) Please, note that for the history of chats, there is "logs browser" with a filtering option. I like this !
However, we can not access to chat history by clicking the user on a private message...
Furthermore, I would like have a field area to write some private notes on my chatfriend.
For instance, I would write : "Eric, 35 years old, single, computer engineer, 2 childs, nice to chat."  It would be a memory aid for occasionnal chatfriends :-)

Thanks for reading.``

PS : The tutorial will follow in 1h. My chatfriend is adding a contents table.

**Comment by Pandore-II** [comment]
### **Adding the harassment report on #metoo** 

### **1) Click on "Scripting", then "Edit events"**
When a private message is received, the program OnQueryMessage is run with parameters as following.

![Ev](https://github.com/kvirc/KVIrc/assets/172998160/5b783914-90d2-4ad6-88e4-a089faef5a18)

The received messages are stored in a variable.
We need to add the following program :

> %DernierMessageRecuEnPv{$0}=%DernierMessageRecuEnPv{$0}    $3
> %UserNameMessageRecuEnPv{$0}=$1
> %HostNameMessageRecuEnPv{$0}=$2

![Evt](https://github.com/kvirc/KVIrc/assets/172998160/7dcda65f-eb83-44ae-9eca-7949b6ffdc9c)

### **2) Click on "Scripting", then "Edit actions", then we modify the program of the button.**

![Btn](https://github.com/kvirc/KVIrc/assets/172998160/84190a74-7e8f-4123-9fc4-3c52923998d4)

The following line will send the report on #metoo quoting the private message received.

> PRIVMSG #metoo :$target (username: %UserNameMessageRecuEnPv{$target} hostname: %HostNameMessageRecuEnPv{$target} realname: $realname($target)) vient d'être bloqué par $me pour message privé non sollicité : %DernierMessageRecuEnPv{$target}

### **3) Exemple of result on #metoo**

![metoo](https://github.com/kvirc/KVIrc/assets/172998160/19cbc490-223c-4026-bbe4-ffaa993f3cc1)


Thanks to Paradoxe80 and some other geeks ;-)

**Comment by johndoe12445** [comment]
Very nice work , it will help to make the site safer 

Keep going  Pandore

**Comment by Yannick80600** [comment]
De rien :)

**Comment by Pandore-II** [comment]
### Au sujet de la blackliste

Messieurs, la blackliste d'Andaris ne fonctionne pas !

At the begging, I tried the following command in the program of my button :

> addbl $target*!*@*

![Blacklist](https://github.com/kvirc/KVIrc/assets/172998160/2002ea49-2186-4988-8057-2a0877927f35)

Private messages are not blocked. It does not work !

So we have to use the list of "registered users"  in Kvirc :

> ignore $target
> reguser.setIgnoreFlags -q -p -n -t -i -d $target

I would write the same comment as previously : there is no search function, the list is not alphabetically sorted.

**Comment by Pandore-II** [comment]
### New feature required : Auto Answer to guest and then, block them.

Guest are people with temporary nickname. They are note good chatfriends and it is (often) useless to chat with them.
I would like to block all of them automatically.

Please have a look at the Andaris and Kvirc configuration :

![Kvric](https://github.com/kvirc/KVIrc/assets/172998160/3fa875f0-6872-45fc-9d4b-c5b6f3ebc254)
![Andaris](https://github.com/kvirc/KVIrc/assets/172998160/4bbb34f8-2d6a-48ff-9917-156d35eca275)

I would like to ask a new feature, as following : 

![KvricBlockGuest](https://github.com/kvirc/KVIrc/assets/172998160/c0b218e5-ff2e-455a-b6a5-aa78cba77807)

**When a guest will send me a private message, he will get a automatic answer before to be blocked.**

At the moment, I think that it is possible to write a program on the event to do this.
![BlockScript](https://github.com/kvirc/KVIrc/assets/172998160/8e3796b1-624f-49a1-be83-a3d23b0e4835)

Someone can help me ?

**Comment by Pandore-II** [comment]
### **New features required : Private note (Kommentar) on user , a quick WHOIS button and a quick log search**

Today Maxhimme sent me a private message. I had already chatted with him but I forgot our chats.

So, I have to go in the "Log viewer", then write his nickname "maxhi" to search...

![Filter](https://github.com/kvirc/KVIrc/assets/172998160/4e0b1b20-203f-4a65-90bf-c6415b4f5ee3)

Then, I can see chat logs :

![LogList](https://github.com/kvirc/KVIrc/assets/172998160/3c6694a9-f245-4cb0-8105-b9963aa8bb69)

As I said, it would be good to be able to write some private notes on chatfriends.... a quick memory aid.

In Kvirc tutorials, I found this possibility ;
https://github.com/kvirc/kvirc.github.io/blob/main/kvirc-tutorials/KVIrc_Registered_Users_de.pdf

![Kommentar](https://github.com/kvirc/KVIrc/assets/172998160/228805a6-9f61-4129-b75a-2533c6f7d3da)

However, I have firslty to add Mahimme to the list of "registered users"... it need a lot of time !

I would ask the following features :

![Exemple](https://github.com/kvirc/KVIrc/assets/172998160/b99e54c8-d119-496d-9003-13bff4820b96)

1) A button to make a quick WHOIS.
   How old is maxhimme  ? Where he live ?

2) A button to show the my private notes about Maxhimme, so the "Kommentar".

3) A button to search quickly all chat history I had with Maxhimme....
It can be good to read it quickly to remember who is Maxhimme.


PS : Please, could you add my tutorial "Guide de survie de la chatteuse IRC"
https://github.com/user-attachments/files/15859936/GuideDeSurvieDeLaChatteuseIRC.pdf
here, https://github.com/kvirc/kvirc.github.io/tree/main/kvirc-tutorials
with other ?

**Comment by ctrlaltca** [comment]
Dear @Pandore-II first thank you for this ticket and your evident enthusiasm in using KVIrc :)
Still, this is a place where users report bugs and feature requests, and this is getting a bit messy at the point where I can't get what you're asking for.
So, I'll try to summarize the requests and answer what i can:

Feature requests:

- a quick button to block an user from sending private messages to you; button should be in the toolbar and also in the right click context menu of the query window
Looks like you already got this working with scripting, very nice. 

- improve the registered users window adding sorting and search 
This is definitely something we should improve. I've created #2654 to track progression on fixing the registered users window.

- a way to see old query logs directly from a query window (toolbar button?)
As far as I know this can't be done with scripting right now. People usually enable the option in Settings Menu -> Configure KVIrc -> IRC -> Query -> Paste last query log
![immagine](https://github.com/kvirc/KVIrc/assets/1631111/c4528a2f-38e6-409b-8ae0-305ef478889e)


- a place to store notes about a chat buddy
Registered users have a "comment" property for this use case.
We need to have a way to identify the user, and that's what registered users are for in KVIrc.
The property can be accessed with scripting: `echo $reguser.property(Nick,comment)`

- auto answer to guest and then, block them.
As far as i know on IRC there's no concept of "guest" user, so maybe it's some specific extension of your IRC network.
If you can identify them eg. by nickname (Guest1, Guest2, ...), you can just add a generic registered user with the mask Guest*!*@* to match them all at once.

- a query toolbar action button to whois the user
You can create an action (Scripting menu -> add action) limited to queries that just executes "whois". This will make the button accessible on the toolbar.

Bug reports:
- Andaris script blacklist doesn't work

Please note that KVIrc is not connected to the the Andaris script team. We don't know them, so you better contact them directly for any problem with this script.

**Comment by Pandore-II** [comment]
Thank for your involvement ctrlaltca.
I just wanted to share my chat experience and to explain the difficulty of being a woman on the chat.
I think that I am not the only one to get problems. The need of making a tutorial of 100 pages to explains how to configure a chat software is already a problem !
I know that I can add some buttons. I spent 1 week to read manuals and create my blocking button. I also asked a lot help on the chat. Before, I had spent several months to chat with Hexchat and Quassel and understand how to configure servers in these software.
I just think that your software should not be reserved only for geeks. Novice users and especially woman should be able to install it quickly and having some preconfigured buttons and tools.

Furthermore, I do not want to complain of  men's behaviors, but without harassment reporting tools, operators can not banish bad users. Thus, behaviors will not change.
If we need to copy-paste each insult or each sexual abuse content and then write a complain to the chat administrator, we will spent so much time.
The other solution is to quit the chat, or to be resigned.
That why, my post was not only about technical features of Kvirc. I just hope that some computer engineers will think about problems of woman on the chat and propose solutions.

Related to my experience, I think that the software must be easy to install, easy to configure, must have by default a lot blocking and reporting tools.
For instance, I am not interested about function related to themes and background images. For me, ergonomics is more important.

Thanks for reading my posts and thanks again you work on these software.

**Comment by DarthGandalf** [comment]
Some meta comments

> Here is also a kvirc tutorial in french

Please don't add many unrelated requests to the same issue on github, that becomes impossible to track and to understand what it's about at all; open separate issues instead about different features/bugs

> So, I tried hexchat... then Quassel... [...] too much for Quassel.

This is relevant how? :) Again, with too many unrelated details it becomes hard to understand what exactly you're asking.

> Au sujet de la blackliste

Since we don't speak French, we have no idea what this chatter is about :) But it does make it harder to understand what this bug is about.

> I just think that your software should not be reserved only for geeks.

I agree, and thanks for feedback!

> And, another button to complain to the chat moderator. As you can see, my button will report the harasser on the #metoo channel.

I don't see how we can solve this in general way in KVIrc side. #metoo is specific to the network you're using, there's nothing in IRC protocol which specifies where harassments should be reported, and such channels don't necessary exist on other networks.

> Guide de survie de la chatteuse IRC

I glanced at the doc, and in first 10 pages it doesn't even mention KVIrc. If someone trying to configure KVIrc discovers the doc, and opens it, they will just immediately close it again, because it doesn't explain what they wanted to learn. I don't see how this can be useful on the KVIrc's website.

---

With so many unrelated discussions and features described in this issue, we likely missed some.

**Comment by DarthGandalf** [comment]
> auto answer to guest and then, block them.

Some networks (e.g. libera.chat) has [user modes](https://libera.chat/guides/usermodes) which doesn't allow users who are not registered in the network to message you: +g and +G. I don't know whether such modes exist on the network you're using, but worth checking.

**Comment by Pandore-II** [comment]
[DarthGandalf](https://github.com/DarthGandalf) , KVIrc is not the finality.

We have only 2 chats in France : chaat.fr and europnet.org
If we need to chat, we will find these websites on google.
Who will search KVIrc to google ?
Then, we start to chat on chaat.fr and europnet.org 
Then we are victim of harassment.... blocked users are not saved on these website....

My tutorial mainly explain this problem and present the best solution I found : KVIrc.

So KVIrc is a tool, and help to avoid to be harassed !

The aim of this tutorial is not KVIrc, but mainly to explain the problem of harassment on chats, and to present the solution !

#metoo on europnet.org is my channel.  Do you think that it is impossible to have a field in the configuration of KVirc to change this ?  for instance #abuse of #support 
I was an example of automatic reporting tools.... Maybe, you will find a better technical solution, I don't know.

I attach you the libre office document. You can occult points which you dislike.

[Guide2.odt](https://github.com/user-attachments/files/15883102/Guide2.odt)

**Comment by Ionic** [comment]
I'd like to point out a few issues with some of the requests:

  - A block button which only discards private messages is probably a bad idea to have in general. Users would most likely expect the button to also discard public channel messages (like the ignore feature does by default). Introducing two buttons - "Ignore" and "Block" would create even more confusion amongst users, since it's not intuitive what each one does. However, adding an "Ignore" button wouldn't satisfy your needs, so you'd have to resort to scripting anyway.
  - Sending automated messages before ignoring/blocking someone or reporting users to a specific channel is a *terrible* idea. This is a huge Denial of Service vector and we certainly don't want to have that in the default configuration. Think about what would happen if a malicious user were to create 100 guest or unregistered users and have them all send a private message to you at the same time - your client would automatically respond with (in your current configuration) 300 total messages to these 100 users, which would lead to *you* being disconnected from the network due to message flood. Easy way to DoS users.
  - The same attack vector naturally exists for automated messages when "reporting" users to a specific channel.

A "quick WHOIS" feature already exists, as far as I remember, as an option in the context menu of a private message window. The `WHOIS` result may only show up in the server log window, but the feature already exists... unless I misremember and scripted that myself, ages ago.

Thanks to @ctrlaltca for creating a bug report for the only improvement request realistically to be implemented upstream - having the registered users list being searchable and sortable by different parameters (like alphabetically or historically - i.e., by the `datetime` the user was added to the list, which is currently the default).

Also, as pointed out earlier, the Andaris script is a third-party tool not affiliated with upstream KVIrc, so your best bet is to contact the original author via the appropriate channels if you want to report bugs or request new features as part of the Andaris script.

**Comment by Pandore-II** [comment]
![RegUser](https://github.com/kvirc/KVIrc/assets/172998160/3ddf6b0e-e5a6-46a4-a951-e49219e1a4ac)

**Comment by Pandore-II** [comment]
### **I am connected on several chats. The list of "registered users" should not be related to the chat connection ?**

If I block the "Axel60" on chaat.fr , are you sure that it is the same user that "Axel60" on europnet.org ?

At the moment, there is only one "registered users" list and user are blocked on all chat servers (or network... I did not understand so much how it works...).

By the way, my buttons will bug if there are several different "Alex" users on several chats...

I think that we should have one list "registered users" list per server... No ?

It does not matter.


### **Here is my new tutorial (in french) to add for 4 buttons **

![SuiteGuide](https://github.com/kvirc/KVIrc/assets/172998160/6fa20c35-42d6-4104-9a4d-fabe6150ed30)

Can be downloaded here :
### **[SuiteGuide.pdf](https://github.com/user-attachments/files/15944367/SuiteGuide.pdf)**
[SuiteGuide.odt](https://github.com/user-attachments/files/15944368/SuiteGuide.odt)

Please note that when a user has been blocked, messages he wrote are saved in the "comment" property of the registered users list.
![BanComment](https://github.com/kvirc/KVIrc/assets/172998160/df09b939-b800-427c-9ecf-5e5370fee677)

Furthermore the harassment report are sent to another server  ##metoo on libera.chat to avoid censorship or sexist behavior from administrator (often men, often sexists).

![MeetooLibera](https://github.com/kvirc/KVIrc/assets/172998160/3c532b05-59d9-4173-b48e-a542799e3e0b)


### The idea of reporting sexual harassment abuses in another server and rating the quality of chat moderation is exposed here :  https://github.com/kiwiirc/kiwiirc/issues/1869

![SafeChat](https://github.com/kvirc/KVIrc/assets/172998160/62391944-78ba-4f80-a11a-03b9e4222071)
![Rating](https://github.com/kvirc/KVIrc/assets/172998160/c96fa57c-f787-408c-bade-b309ae192c8a)


Thanks for reading ;-)

**Comment by Pandore-II** [comment]
Sorry for this new post but I would like to have also a button to change the color of text... like in KVIrc ;-)

![ColorBtn](https://github.com/kvirc/KVIrc/assets/172998160/fc322cf9-e1fd-42e2-9913-e6a4d9a8137e)



Thanks for reading.

**Comment by Pandore-II** [comment]
### And about the history of chats :

![SortChatHistory](https://github.com/kvirc/KVIrc/assets/172998160/3a38c5e6-8d85-49db-a29c-2249a9f4436d)

Thanks for reading.
So sorry for this new post...

**Comment by DarthGandalf** [comment]
@Pandore-II could you please report separate issues as separate issues, and not dump everything together? It's really hard otherwise to track what's done, and what's not

**Comment by Pandore-II** [comment]
Ok [DarthGandalf](https://github.com/DarthGandalf)

However, I don't know what is relevant for you... I just wrote here some idea... Sometimes, I made some errors (for instance with the Black list).

I added here a new issue :
https://github.com/kvirc/KVIrc/issues/2655

**Comment by DarthGandalf** [comment]
Thanks. Is that the only feature you needed?

You see, when the idea is wrong, the separate issue about that idea can simply be closed, and no harm is done. However, when the wrong idea is dumped at the same place as many other issues and feature requests, it destroys value of those, because it'll be hard to find anything useful in this long text anymore, if some of ideas listed need to be randomly ignored.

**Comment by Pandore-II** [comment]
Ok @DarthGandalf 

Yesterday the chat coco has been closed by police ( https://www.leparisien.fr/faits-divers/coco-le-site-mis-en-cause-dans-des-guets-apens-homophobes-ferme-par-les-aurorites-francaises-25-06-2024-TDV3FOUR4NBODN2PILJ4NTZSX4.php )
As you can see, my post here was very useful ;-)
Is is an European police inquiry, you will find other press paper in English. Sorry to quote this one.

The most important ideas are not so technical issues.
I shared my feedback and underlined the problem of sexual harassment, safety of women on chats, difficulties to block bad guys, problems of sexist server administrator...

Some of these problem are directly related to the ergonomic of software tools. Other needs to be thought, to propose solution (rating of servers, reputation of users, sharing list of blocked users, creating statistics about abuses, collecting complains). It is mainly for this, I posted here...

I am attached to the liberty and democratic values but some people need to be protected... chat moderation should be improved before that police close the chats...
Often, women are the main victims....
